src/solvers/strings/array_pool.cpp

```diff
@@ -184,9 +184,12 @@ void array_poolt::insert(
   attempt_assign_length_from_type(array_expr, length_of_array, fresh_symbol);
 }
 
-const std::set<array_string_exprt> &array_poolt::created_strings() const
+/// Return a map mapping all array_string_exprt of the array_pool to their
+/// length.
+const std::unordered_map<array_string_exprt, exprt, irep_hash> &
```

Comment:
:bulb: exposing the whole map here seems unnecessary since we are only using the first component. An alternative would be to define a `for_each_string` function with takes a function as argument and apply it to each `array_string_exprt`.

Reply:
I'll leave it like this for now so that the PR doesn't blow up.